src/lib/hooks/use-token-history.ts

```diff
@@ -0,0 +1,42 @@
+import { useQuery } from '@tanstack/react-query'
+import { AssetTransfersWithMetadataResult } from 'alchemy-sdk'
+import { Address } from 'viem'
+import { useAccount } from 'wagmi'
+
+export const useTokenHistory = (...tokens: (string | Address)[]) => {
+  const { address: user, chainId } = useAccount()
```

Comment:
Please use `useWallet` for use account like data: https://github.com/IndexCoop/index-app/blob/master/src/lib/hooks/use-wallet.ts